### PR TITLE
[BUGFIX Fix Alignment setting

### PR DESCRIPTION
**Pull request by DominikBucher12**
# What this PR do:
- Fixes alignment setting so now you can set alignment to each line independently :) 
# How to test it
- Run app try to mimc same behaviour as in the video or some more.
# Where you can start
-  If the tests are running, you are fine :) 

https://github.com/danielsaidi/RichTextKit/assets/17381941/bdc4a5f8-bc34-407c-89b2-769175b4a012

### CODE REVIEW

File: Sources/RichTextKit/Component/RichTextViewComponent+Alignment.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ public extension RichTextViewComponent {
         _ alignment: RichTextAlignment
     ) {
         if richTextAlignment == alignment { return }
-        if !hasTrimmedText {
+        if !hasTrimmedText || !hasSelectedRange {
```
Comment (DominikBucher12) [inline Sources/RichTextKit/Component/RichTextViewComponent+Alignment.swift:36]: We should make this function call setAttributes which decides internally if has selected range or not. But that is music of the future.

**Comment by DominikBucher12** [comment]
Screenshot 2024-01-24 at 12 55 59

Also currently on macOS since no configuration with default font is set yet.

**Comment by danielsaidi** [comment]
Great job! Merging this to give it a go. 😀

**Comment by danielsaidi** [comment]
@DominikBucher12 This doesn't work for me, see movie. The same problem on both macOS and iOS.

**Comment by danielsaidi** [comment]
https://github.com/danielsaidi/RichTextKit/assets/429927/c8840395-3c42-4b3b-8a84-8334afeaa818

**Comment by danielsaidi** [comment]
If you want to continue to look at this, I merged it to `paragraph-fix`.

**Comment by danielsaidi** [comment]
@DominikBucher12 Actually, if I just have this in `RichTextViewComponent+Alignment` it behaves better than now, at least on macOS:

```swift
public extension RichTextViewComponent {

    /// Get the rich text alignment at current range.
    var richTextAlignment: RichTextAlignment? {
        guard let style = richTextParagraphStyle else { return nil }
        return RichTextAlignment(style.alignment)
    }

    /// Set the rich text alignment at current range.
    ///
    /// > Todo: Something's currently off with alignment. It
    /// spils over to other paragraphs when moving the input
    /// cursor and inserting new text.
    func setRichTextAlignment(
        _ alignment: RichTextAlignment
    ) {
        if richTextAlignment == alignment { return }
        setRichTextAlignment(alignment, at: selectedRange)
    }
}
```

**Comment by danielsaidi** [comment]
I have pushed a MUCH shorter version of the alignment setter to the new branch. It seems to work pretty well, but please check it out too.

**Pull request merged**